dashboard/test/models/vocabulary_test.rb

```diff
@@ -25,4 +25,55 @@ class VocabularyTest < ActiveSupport::TestCase
       vocab.save!
     end
   end
+
+  test 'vocabulary automatically sanitizes word when turning it into key' do
+    vocab = create :vocabulary,
+      word: 'Some !! ßtring 123 ,with, "ILLEGAL" _characters_.',
+      key: nil
```

Comment:
why is it necessary to specify key here, given that it gets auto generated from the word?

Reply:
the factory attempts to generate its own key rather than generating it from the word; passing in `nil` here prevents the factory from doing so which allows us to actually use the "generate from word" bit